src/pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -334,7 +334,9 @@ def _print_results(results: List[_OUT_DICT], stage: str) -> None:
 
         # fallback is useful for testing of printed output
         term_size = shutil.get_terminal_size(fallback=(120, 30)).columns or 120
-        max_length = int(min(max(len(max(metrics_strs, key=len)), len(max(headers, key=len)), 25), term_size / 2))
+        # encode as gbk, which treat Chinese characters as twice the width of alphabet characters.
+        matrics_encoded = [metric.encode("gbk") for metric in metrics_strs]
```

Comment:
Can we please use an encoding that is not specialized on chinese characters? GBK specializes on chinese characters and performs worse for others. I'd prefer something like UTF-8, UTF-16 or even UTF-32 (depending on what we want to support) instead

Reply:
Yes you are right, gbk makes sure Chinese and English characters work as expected. However, I didn’t try other languages. I will have a look and see if other encodings can make it work.

Reply:
Hi @justusschock I searched for methods to handle other languages, but it seems that it is not a trivial task. 

I found this repo useful: https://github.com/fedora-infra/kitchen/blob/73e09ff3ac6e453f1e6f307cb2a48474b7435f82/kitchen3/kitchen/text/display.py#L335

It seems that it can properly calculate the character width for Chinese, Japanese, Korean and Spanish at least. I haven't tested other languages, but I am not sure if it is necessary to support more languages.

Reply:
@heng-yuwen thanks for looking into this. I do not quite understand the difficulty here.
From looking at the utf-8 encoding table [here](https://www.utf8-chartable.de/unicode-utf8-table.pl) it seems that utf-8 does support encoding for chinese, english and most other languages (including cyrillic alphabets and others as well).
Why can't we just use this?

Reply:
@justusschock The problem is not about encoding, but about how to get the printed character width. If you try to encode the Chinese character as utf-8, the len() function gives the character number instead of width.

Reply:
Also, if you are interested, gbk encodes Chinese characters with 2 bytes, which is also the length for most Chinese characters. That's why the length of gbk encoding can make Chinese characters look fine when printed. For other languages, it needs extra dedicated way to deal with the length.

Reply:
If you write a test that shows the bug/issue, then we will understand the problem better so that we can find a reliable solution across languages